src/memory/global_memory_control.cpp

```diff
@@ -118,10 +120,14 @@ static bool my_commit(extent_hooks_t *extent_hooks, void *addr, size_t size, siz
     return err;
   }
 
+  [[maybe_unused]] auto blocker = memgraph::utils::MemoryTracker::OutOfMemoryExceptionBlocker{};
   if (GetQueriesMemoryControl().IsThreadTracked()) [[unlikely]] {
-    GetQueriesMemoryControl().TrackAllocOnCurrentThread(length);
+    bool ok = GetQueriesMemoryControl().TrackAllocOnCurrentThread(length);
+    DMG_ASSERT(ok);
```

Comment:
So, the logic here is to assert we cannot fail on commit. I wouldn't agree with that logic, as we can fail, and then we will crash the database. I would rather remove DMG_ASSERT, leave OOMBlocker as is, and catch OOM on next allocation via alloc hook

Reply:
1. DMG_ASSERT is only for a debug build, so production release builds are never effected
2. As setup ATM, OOMBlocker should provide the guarantee that this will always be true.

Reply:
Makes sense